dlk/backends/include/cpp.h

```diff
@@ -90,28 +69,26 @@ void qconv(T_q in_data_packed[], T_out out_data[], T_q k_data_packed[], unsigned
     cpp::qconv1x1_impl(in_data_packed, out_data, k_data_packed, in_w, in_h, in_c_by_word, out_c);
   }
 }
+/// @}
 
-template <class Ta, class Tb, class Ty>
-void gemm(Ta A[], Tb B[], Ty Y[], u32 a_row, u32 a_col, u32 b_col);
-
-template <class Tx, class Ty, int Nbit_a, int Nbit_b>
-void qgemm(Tx A_packed[], Tx B_packed[], Ty Y[], u32 a_row, u32 a_col, u32 b_col);
-
-void qconv_with_kn2row_impl(T_q in_data[], T_out out_data[], T_q k_data[], T_out partial_out_data[],
-                            T_out threshold_data[], unsigned in_w, unsigned in_h, unsigned in_c, unsigned out_w,
-                            unsigned out_h, unsigned out_c, unsigned k_w, unsigned k_h, unsigned pad);
+/// convolution with kn2row and tiling
+/// @{
+void conv_kn2row_tiling_impl(T_in in_data[], T_out out_data[], T_k k_data[], T_out threshold_data[],
+                             const unsigned in_w, const unsigned in_h, const unsigned in_c, const unsigned out_w,
+                             const unsigned out_h, const unsigned out_c, const unsigned k_w, const unsigned k_h,
+                             const unsigned pad, const unsigned stride);
 
 template <int KH, int KW>
-void qconv_with_kn2row(T_q in_data_packed[], T_out out_data[], T_q k_data_packed[], T_out threshold_data[],
-                       unsigned in_w, unsigned in_h, unsigned in_c_by_word, unsigned nbits_in_data, unsigned out_w,
-                       unsigned out_h, unsigned out_c, unsigned pad, unsigned stride)
+void conv_kn2row_tiling(T_in in_data[], T_out out_data[], T_k k_data[], T_out threshold_data[], const unsigned in_w,
+                        const unsigned in_h, const unsigned in_c, const unsigned out_w, const unsigned out_h,
+                        const unsigned out_c, const unsigned pad, const unsigned stride)
 {
   assert(((KH == 3) && (KW == 3)) || ((KH == 1) && (KW == 1)));
   assert(((KH == 3) && (pad == 1)) || ((KH == 1) && (pad == 0)));
   assert(stride == 1);
 
-  qconv_with_kn2row_impl(in_data_packed, out_data, k_data_packed, out_data, threshold_data, in_w, in_h, in_c_by_word,
-                         out_w, out_h, out_c, KH, KW, pad);
+  cpp::conv_kn2row_tiling_impl(in_data, out_data, k_data, threshold_data, in_w, in_h, in_c, out_w, out_h, out_c, KW, KW,
+                               pad, stride);
 }
 
 void qconv_kn2row_tiling_impl(T_q in_data[], T_out out_data[], T_q k_data[], T_out threshold_data[],
```

Comment:
Should be asking in the previous PR, but what is the reason for splitting implementation into `qconv_with_kn2row` and `qconv_with_kn2row_impl`, considering that we have only *one* implementation? :thinking: